src/game.cpp

```diff
@@ -1830,9 +1832,21 @@ void game::set_critter_died()
 
 void game::update_weather()
 {
+    w_point &w = *weather_precise;
+    int old_wind_direction = w.winddirection;
+    int old_wind_speed = w.windpower;
+    if( wind_direction_override != -1 ) {
+        winddirection = wind_direction_override;
+    } else {
+        winddirection = old_wind_direction;
+    }
+    if( windspeed_override != -1 ) {
+        windspeed = windspeed_override;
+    } else {
+        windspeed = old_wind_speed;
+    }
```

Comment:
This seems way too verbose. Can be shortened down to two lines of code:
```c++
winddirection = wind_direction_override ? *wind_direction_override : weather_precise->winddirection;
windspeed = windspeed_override ? *windspeed_override : weather_precise->windpower;
```
BTW own getters for these values would be better: like `get_wind_speed()` that takes overrides into account.